addons/udes_stock/models/stock_picking.py

```diff
@@ -1235,10 +1254,15 @@ def get_suggested_locations(self, move_line_ids):
 
         for picking in self:
             policy = picking.picking_type_id.u_drop_location_policy
-
             if policy:
-                func = getattr(self, '_get_suggested_location_' + policy, None)
-
+                default_dest = picking.picking_type_id.default_location_dest_id
+                if move_line_ids and all(ml.location_dest_id != default_dest
+                                         for ml in move_line_ids):
+                    # If we have pre selected a location use it
+                    func = self._get_suggested_location_exactly_match_move_line
```

Comment:
Not sure this will work, but I don't think I have the full picture of what you're trying to achieve.

The allocation during the assignment trivially uses the first suggested location of the configured policy. This does not prevent a given location being used for multiple move line assignments. So, at this point, using the `exactly_match_move_line` policy may (likely) result in a bad suggestion.

Reply:
As agreed by voice, consider:
 - checking the pre-process flag as part of the if clause;
 - factor out the move line loc dest check to be a move line method and document the assumptions, just for the sake of readability.